### DISCUSSION TITLE
Section Properties from a Shapley Polygon

### DISCUSSION THREAD

**Discussion by nira123** [robbievanleeuwen/section-properties#239] [Q&A] [answered]
Thank you for developing this module. I am quit new here please bare with me.

I am trying to get the section properties from a Shapley Polygon including the torsional constant. 
I am getting the flowing error:
ValueError: Argument 'mesh_sizes' for a Geometry must be either a float, or a list of float with length of 1, not [0.0005, 0.001].

Can someone please help. 
Code is given below:

```python
from sectionproperties.pre.geometry import Geometry
from sectionproperties.analysis.section import Section
from sectionproperties.pre.pre import Material
from sectionproperties.analysis.section import Section
from shapely.geometry import Polygon, LineString, Point

precast = Material(
    name="65 MPa",
    elastic_modulus=37.4e3,
    poissons_ratio=0.2,
    yield_strength=65,
    density=2.4e-6,
    color="grey",
)

poly_2=[(-1000, 1150),
 (1000, 1150),
 (1000, 1300),
 (-1000, 1300)]


p2 = Polygon(poly_2)
geometry = Geometry(p2, material=precast)
geometry.create_mesh(mesh_sizes=[0.0005, 0.001])
section = Section(geometry)
section.plot_mesh()  # plot the generated mesh
section.calculate_geometric_properties()
section.calculate_warping_properties()
section.calculate_plastic_properties()
section.plot_centroids()
```

**Comment by nira123** [comment, marked as answer]
I think I found the error :
geometry.create_mesh(mesh_sizes=[500])
Thanks. 
Is it possible to filter only J?

**Reply by nira123** [reply]
Got it :
section.get_j()
Awesome thanks.

**Comment by Czarified** [comment]
Nira,

Glad you worked it out! The mesh_sizes argument does accept a list, if you have multiple meshing regions defined. I believe we have an example of this in the docs.

the `.get_j()` is probably the easiest way to grab only the torsional constant.

**Reply by nira123** [reply]
Cheers Czarified.
Thanks a lot.

**Comment by robbievanleeuwen** [comment]
Hi @nira123, welcome to *sectionproperties*. Glad you sorted your issue out, let us know if you have any more questions!

**Reply by nira123** [reply]
Thanks a lot Robbie.
Excellent work!
